platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/AccountsManager.java

```diff
@@ -53,23 +105,26 @@ private String validateUserId() {
   }
 
   private void validateRotation(String userId) {
-    SharedPreferences sharedPreferences = getSharedPreferences();
-    timestamp = sharedPreferences.getLong(PREFERENCE_TIMESTAMP, 0L);
-    skuToken = sharedPreferences.getString(PREFERENCE_SKU_TOKEN, "");
-    if (timestamp == 0L || TextUtils.isEmpty(skuToken)) {
+    if (TextUtils.isEmpty(skuToken) || timestamp == 0L) {
       skuToken = generateSkuToken(userId);
       timestamp = persistRotation(skuToken);
     }
   }
 
+  @Nullable
   String getSkuToken() {
-    if (isExpired()) {
+    if (isManaged) {
+      if (isExpired()) {
+        SharedPreferences sharedPreferences = getSharedPreferences();
+        String userId = sharedPreferences.getString(PREFERENCE_USER_ID, "");
+        skuToken = generateSkuToken(userId);
+        timestamp = persistRotation(skuToken);
+      }
+    } else {
       SharedPreferences sharedPreferences = getSharedPreferences();
-      String userId = sharedPreferences.getString(PREFERENCE_USER_ID, "");
-      skuToken = generateSkuToken(userId);
-      timestamp = persistRotation(skuToken);
+      String notManagedSkuToken = sharedPreferences.getString(KEY_PREFERENCE_SKU_TOKEN, "");
+      skuToken = notManagedSkuToken;
```

Comment:
You can verify if the `notManagedSkuToken` is not `null` before assigning it to the `skuToken` (otherwise assign `""`) to make the `#getSkuToken` non-null and effectively the `skuToken` field as well.

Reply:
I think that extra check is not necessary. According to the docs https://developer.android.com/reference/android/content/SharedPreferences.html#getString(java.lang.String,%20java.lang.String):
> Returns the preference value if it exists, or defValue.

which in this case would be empty `""` (not `null`).

The IDE is also warning me about that:
> Condition 'notManagedSkuToken == null' is always 'false'

Reply:
Cool. I just noticed that ["passing null for this argument is equivalent to calling remove(java.lang.String) with this key"](https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#putString(java.lang.String,%20java.lang.String)) so it makes sense.